co-log-core/src/Colog/Core/Class.hs

```diff
@@ -18,11 +17,13 @@ TODO: laws
 class HasLog env msg m where
     getLogAction :: env -> LogAction m msg
     setLogAction :: LogAction m msg -> env -> env
+    setLogAction = overLogAction . const
+    {-# INLINE setLogAction #-}
+    overLogAction :: (LogAction m msg -> LogAction m msg) -> env -> env
+    overLogAction f env = setLogAction (f $ getLogAction env) env
+    {-# INLINE overLogAction #-}
+    {-# MINIMAL getLogAction, (setLogAction | overLogAction) #-}
 
 instance HasLog (LogAction m msg) msg m where
     getLogAction = id
-    setLogAction = const
-
-overLogAction :: HasLog env msg m => (LogAction m msg -> LogAction m msg) -> env -> env
-overLogAction over env = setLogAction (over $ getLogAction env) env
-{-# INLINE overLogAction #-}
+    overLogAction = id
```

Comment:
Let's also implement `setLogAction` function. If you don't implement methods with default definitions from typeclass, then the default implementation of those methods is used. If you implement methods manually, you have better control over the implementation and thus you can have better performance. Also, let's add `{-# INLINE #-}` pragmas to all three functions in this instance to have even more guarantees for performance.